server/src/userManagement.mjs

```diff
@@ -197,6 +199,9 @@ function addUser(userId) {
     ws.on('pong', async hb => {
       heartbeat(ws)
     });
+    ws.on('close', function close() {
+      user2ws.delete(''+userId);
```

Comment:
I think we need to do this a little differently. This line deletes the entire Array of WS objects for the user. If we run this when the user has more than one connection the others will be put in a weird state.

We should go through the array and delete _only_ the current ws object from the array.